exporter/newrelicexporter/newrelic.go

```diff
@@ -171,39 +104,28 @@ func (e *exporter) extractInsertKeyFromHeader(ctx context.Context) string {
 	return values[0]
 }
 
-func (e *exporter) pushTraceData(ctx context.Context, td pdata.Traces) (outputErr error) {
-	var (
-		errs      []error
-		sentCount int
-	)
+func (e exporter) pushTraceData(ctx context.Context, td pdata.Traces) (outputErr error) {
+	details := newTraceMetadata(ctx)
+	details.dataInputCount = td.SpanCount()
+	builder := func() (telemetry.PayloadEntry, error) { return e.buildTracePayload(&details, td) }
+	return e.export(ctx, &details, builder)
 
-	startTime := time.Now()
-	insertKey := e.extractInsertKeyFromHeader(ctx)
+}
 
-	details := newTraceMetadata(ctx)
-	defer func() {
-		apiKey := sanitizeAPIKeyForLogging(insertKey)
-		if apiKey != "" {
-			details.apiKey = apiKey
-		}
-		details.dataOutputCount = sentCount
-		details.exporterTime = time.Since(startTime)
-		details.grpcResponseCode = status.Code(outputErr)
-		err := details.recordMetrics(ctx)
-		if err != nil {
-			e.logger.Error("An error occurred recording metrics.", zap.Error(err))
-		}
-	}()
+func (e exporter) buildTracePayload(details *exportMetadata, td pdata.Traces) (telemetry.PayloadEntry, error) {
```

Comment:
Woops gotta change `*exportMetadata` to `exportMetadata`.

Reply:
Wouldn't that mean we wouldn't be able to modify attributes in the `details`? 🤔

Reply:
Actually, maybe not. `export` has to have a pointer for `exportMetadata`  because it needs to change its values.

I think that suggests we should use pointers to `*exportMetadata` throughout for consistency.